### PR TITLE
Augment witnessctl update support to know about FT

### PR DESCRIPTION
**Pull request by AlCutter**
This is an outline of adding FT support to the update-applet `witnessctl` command.

It's mainly just to bring this code (somewhat) along with the self-update changes happening at the moment.

### CODE REVIEW

File: api/api.proto [hunk 1]
```diff
@@ -73,14 +73,42 @@ AppletUpdate
 A `AppletUpdate` represents an OTA sequence applet slice.
 
 The `TotalChunks` value indicates the total number of chunks for the update,
-`Seq` is the transmitted `Data` chunk number (starting with 1 as the value for
-the first chunk).
+`Seq` is the transmitted AppletUpdate chunk number:
+ - `0` indicates that the struct contains verification data in `Header`.
+ - `1` onwards identifies the first, second, ... chuck of firmware image data.
```
Comment (mhutchinson) [inline api/api.proto:78]:
```suggestion
 - `1` onwards identifies the first, second, ... chunk of firmware image data.
```

File: api/api.proto [hunk 2]
```diff
@@ -73,14 +73,42 @@ AppletUpdate
...
+An `AppletUpdateHeader` contains verification material for an applet update.
+It's sent in the zero-th `AppletUpdate` message during an OTA.
+
+
+*/
+
+message AppletUpdateHeader {
+	// Signature holds the signature over the applet.
+	bytes  Signature  = 1;
+	// Checkpoint contains a note-formatted LogCheckpoint.
```
Comment (mhutchinson) [inline api/api.proto:104]:
```suggestion
	// Checkpoint contains a note-formatted Log Checkpoint.
```

File: api/api.proto [hunk 3]
```diff
@@ -73,14 +73,42 @@ AppletUpdate
...
+
+An `AppletUpdateHeader` contains verification material for an applet update.
+It's sent in the zero-th `AppletUpdate` message during an OTA.
+
+
+*/
+
+message AppletUpdateHeader {
+	// Signature holds the signature over the applet.
+	bytes  Signature  = 1;
```
Comment (mhutchinson) [inline api/api.proto:103]: Whose signature is this? Isn't this already in the Manifest?

Comment (AlCutter) [reply api/api.proto:103]: This is just the existing signature-over-applet made a bit more explicit (it was previously smuggled in via `Data` if `Seq == 0`).

I'd expect it's the manifest itself which will (eventually) be signed - I guess we could use a `note` format for that and have it "attached" if we wanted), or this `Signature` field could be repurposed for that.

**Review by mhutchinson — APPROVED** [review]

**Pull request merged**